### ISSUE TITLE
Simplify x**2 / abs(x) to abs(x) if x is real

### ISSUE THREAD

**Issue by maltaisn** [sympy/sympy#20108]
I was surprised to find the following result:
```python
x = Symbol('x', real=True)
simplify(x**2 / abs(x))
```
```text
x**2/Abs(x)
```
Shouldn't it simplify to `Abs(x)` given that `x` is real?

My current workaround is:
```
e = e.subs({s**2/abs(s): abs(s) for s in e.free_symbols if s.is_real})
```

**Comment by BhaskarJoshi-01** [comment]
@oscarbenjamin  Claim

**Comment by sylee957** [comment]
Can this be generalized to higher powers or multiplication with other terms
In our point of simplify, it should be rewriting `Mul` opportunistically only when there is an exact occurance of `abs(s) / s ** 2` which can be very limited

**Comment by BhaskarJoshi-01** [comment]
@maltaisn Your approach is completely fine. I think you should change the code to 

 ```
def _abs_simplify(e: Expr) -> Expr:
    subs = {}
    for s in e.free_symbols:
        if s.is_real and not s.is_nonnegative:
            subs[abs(s) / s ** 2] = 1 / abs(s)
            subs[s ** 2 / abs(s)] = abs(s)
    return e.subs(subs)
```
I think this should work fine.

**Comment by dhruvmendiratta6** [comment]
> In our point of simplify, it should be rewriting `Mul` opportunistically only when there is an exact occurance of `abs(s) / s ** 2` which can be very limited

But 
```
x = Symbol('x', real = True)
print(diff(Abs(x**3),x))
```
gives `x**2*sign(x) + 2*x*Abs(x)`
I think here `Abs(x**3)` is converted to `x**2*Abs(x)` and then differentiated. Here `Abs(x**2)` is converted to `x**2` Wouldn't it make sense to also expect this kind of behavior from simplify?

**Comment by sylee957** [comment]
Nevermind, I thought that `subs` can only detect the exact patterns, but it may generalize to `x**3 / abs(x)` or `2*x**2 / abs(x)` or such with some internal heuristics.
However, I'd want to identify the clear scope about how much generality this can solve problems outside of the exact patterns.

**Comment by dhruvmendiratta6** [comment]
> However, I'd want to identify the clear scope about how much generality this can solve problems outside of the exact patterns.

If by exact patterns you mean some power of `x` along with `Abs(x)`, I can't think of any other.

**Comment by sylee957** [comment]
Yes, and the other problem is that x can be anything other than the exact symbol, but any arbitrary symbol

**Comment by dhruvmendiratta6** [comment]
After thinking about it and trying to make it work for an arbitrary expression, I don't think it is generalisable.

**Comment by vorabrijesh** [comment]
We cannot  say x**2 /abs(x) = abs(x) as  Left hand side is not defined on x=0 and right hand side is defined. Both the LHS and RHS are different functions and have different graphs

**Comment by oscarbenjamin** [comment]
At least this could simplify (note that nonzero implies real):
```python
In [3]: x = Symbol('x', nonzero=True)

In [4]: simplify(x**2/Abs(x))
Out[4]: 
  2
 x 
───
│x│
```

**Comment by pallosp** [comment]
I encountered a similar missed opportunity:

`x * sign(x)` could be simplified to `Abs(x)` when `x` is real.

**Comment by mayank21singh** [comment]
Hi, I would like to work on this. I plan to look into simplify or powsimp to handle the x**2 / Abs(x) case for real symbols.

**Comment by haru-44** [comment]
We might want to add rewriting rules for `Abs` such as the following:

* `x**(2*k)` -> `Abs(x)**(2*k)`
* `sign(x)` -> `Abs(x)/x` if `x` is not zero

**Comment by mayank21singh** [comment]
> We might want to add rewriting rules for `Abs` such as the following:
> 
> * `x**(2*k)` -> `Abs(x)**(2*k)`
> * `sign(x)` -> `Abs(x)/x` if `x` is not zero

Thanks @haru-44! I have implemented the rewrite rules as suggested.

sign(x).rewrite(Abs) now returns x/Abs(x) if x is nonzero, otherwise it defaults to the Piecewise form.

x**(2*k).rewrite(Abs) now returns Abs(x)**(2*k) (unevaluated) for real x and even powers.

I have also added comprehensive tests for both cases in test_complexes.py and test_power.py.
